Assets/Scripts/FrameWork/WorldTime/WorldTime.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+public class WorldTime : MonoBehaviour
+{
+    private int _day;
+    private float _time;
+    [Range(0, 23)] [SerializeField] private int startHour = 6;
+
+    [Header("Day/Night Cycle Settings")] 
+    [Range(0, 23)] [SerializeField] private int dayStartHour = 6;
+    [Range(1, 24)] [SerializeField] private int endOfDayHour = 24;
+    
+    [Space]
+    [Range(1, 120)] [SerializeField] private int dayInMinutes = 1;
+    private int _timeMultiplier;
+    [SerializeField] private bool useRealTime;
+
+    [Header("UI Settings")]
+    [SerializeField] private bool useUI;
+    [SerializeField] private Text worldClock;
+    [SerializeField] private bool showSeconds;
+
+    [Header("Unity Events")] 
+    public UnityEvent onNewDay;
+
+    private void Start()
+    {
+        InitClock();
+    }
+
+    private void InitClock()
+    {
+        _time = startHour;
+        if (!useUI) return;
+        worldClock = GameObject.Find("WorldClock").GetComponent<Text>();
+    }
+
+    public int CurrentDay => _day;
+    public float CurrentTime => _time;
+    public int DayInMinutes => dayInMinutes;
+    public bool IsEndOfDay => (int)_time == endOfDayHour;
```

Comment:
Can this be above start or under all the functions?

Reply:
:o